File: Algorithms/Implementation/BeautifulDaysAtTheMovies.java
package Implementation;

import java.util.Scanner;

/**
 *	Algorithms -> Implementations -> Beautiful Days at the Movies
 *	Easy
 */
public class BeautifulDaysAtTheMovies {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		
		int startDay = in.nextInt();
		int endDay = in.nextInt();
		int k = in.nextInt();
		
		in.close();
		
		int count = calculateTheNumberOfBeautifulDays(startDay, endDay, k);
		System.out.println(count);
	}

	public static int calculateTheNumberOfBeautifulDays(int startdate, int enddate, int divisibilityFactor)
	{
		int count = 0;
		
		for(int day = startdate; day <= enddate; day++) {
			if(isBeautifulDayModK(day, divisibilityFactor)) count++;
		}
		
		return count;
	}
	
	public static boolean isBeautifulDayModK(int day, int k)
	{
		int reversed =  reverseInteger(day);
		return (day - reversed) % k == 0;
	}
	
	public static int reverseInteger(int number)
	{
		int reversed = 0;
		
		char[] numToString = String.valueOf(number).toCharArray();
		char[] reversedArray = new char[numToString.length];
		
		for(int i = 0; i < numToString.length; i++) {
			reversedArray[numToString.length - i - 1] = numToString[i];
		}
		
		reversed = Integer.parseInt(String.valueOf(reversedArray));
		
		return reversed;
	}
}

File: Algorithms/Implementation/BeautifulDaysAtTheMovies.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

/**
 *	Algorithms -> Warmup -> Beautiful Days at the Movies
 *	Easy
 */

// Takes an integer and returns the reverse of it
int reverseInteger(int number) {
	int reverse = 0;
	int n = number;
	while (n != 0) {
		reverse = reverse * 10;
	    reverse = reverse + n % 10;
	    n = n / 10;
	}
	//printf("reverse: %d\n", reverse);
	return reverse;
}

//Checks if a day is beautiful mod k
int isDayBeautiful(int number, int mod) {
	int result = 0;
	//printf("abs(number - reverseInteger(number)) mod: %d\n", abs(number - reverseInteger(number)) % mod);
	if(abs(number - reverseInteger(number)) % mod == 0) {
		result = 1;
	}
	return result;
}

int main(){
	int startDay, endDay, divisibilityFactor;

	scanf("%d %d %d", &startDay, &endDay, &divisibilityFactor);

	int beautifulCount = 0;

	for(int day = startDay; day <= endDay; day++) {
		beautifulCount += isDayBeautiful(day, divisibilityFactor);
	}

	printf("%d", beautifulCount);

    return 0;
}

